srcCpp/perftest_publisher.cxx

```diff
@@ -1192,14 +1200,9 @@ int perftest_cpp::Subscriber()
                 if (showCpu) {
                     outputCpu = reader_listener->cpu.get_cpu_instant();
                 }
-                printf("Packets: %8llu  Packets/s: %7llu  Packets/s(ave): %7.0lf  "
-                       "Mbps: %7.1lf  Mbps(ave): %7.1lf  Lost: %5llu (%1.2f%%) %s\n",
-                        last_msgs, mps, mps_ave,
-                        bps * 8.0 / 1000.0 / 1000.0, bps_ave * 8.0 / 1000.0 / 1000.0,
-                        reader_listener->missing_packets,
-                        missing_packets_percent,
-                        outputCpu.c_str()
-                );
+                _printer.print_throughput(last_msgs, mps, mps_ave, bps, bps_ave,
```

Comment:
```
_printer.print_throughput(
        last_msgs,
        mps,
        mps_ave,
        bps,
        bps_ave,
        reader_listener->missing_packets,
        missing_packets_percent,
        outputCpu.c_str());
```

Reply:
I am wondering if you could evaluate doing the following (see if it makes sense):

- Make the `get_cpu_instant()` generate a number (`float` or `double`) instead of a `string`
- Modify the outPutCpu variable to be a `double` or a `float`
- Modify the `print_throughput()` to receive the double or float and convert it with the right precission to a string.

Reply:
ok